packages/language-service/src/lib/service.ts

```diff
@@ -295,41 +294,59 @@ export class StylableLanguageService {
-                              new ProviderPosition(
-                                  position.line,
-                                  Math.max(position.character - 1, 0)
-                              ),
-                              position
-                          ),
-                    typeof com.insertText === 'string' ? com.insertText : com.insertText.source
+        const groupedCompletions = new Map<string, CompletionItem>();
+
+        for (const stComp of stCompletions) {
+            const lspCompletion = CompletionItem.create(stComp.label);
+            const ted: TextEdit = TextEdit.replace(
+                stComp.range
+                    ? stComp.range
+                    : new ProviderRange(
+                          new ProviderPosition(position.line, Math.max(position.character - 1, 0)),
+                          position
+                      ),
+                typeof stComp.insertText === 'string' ? stComp.insertText : stComp.insertText.source
+            );
+            lspCompletion.insertTextFormat = 2;
+            lspCompletion.detail = stComp.detail;
+            lspCompletion.textEdit = ted;
+            // override sorting order to the top
+            // todo: decide on actual sorting for all stylable completions
+            lspCompletion.sortText = 'a';
+            lspCompletion.filterText =
+                typeof stComp.insertText === 'string'
+                    ? stComp.insertText
+                    : stComp.insertText.source;
+            if (stComp.additionalCompletions) {
+                lspCompletion.command = Command.create(
+                    'additional',
+                    'editor.action.triggerSuggest'
                 );
-                lspCompletion.insertTextFormat = 2;
-                lspCompletion.detail = com.detail;
-                lspCompletion.textEdit = ted;
-                lspCompletion.sortText = com.sortText;
-                lspCompletion.filterText =
-                    typeof com.insertText === 'string' ? com.insertText : com.insertText.source;
-                if (com.additionalCompletions) {
-                    lspCompletion.command = Command.create(
-                        'additional',
-                        'editor.action.triggerSuggest'
-                    );
-                } else if (com.triggerSignature) {
-                    lspCompletion.command = Command.create(
-                        'additional',
-                        'editor.action.triggerParameterHints'
-                    );
-                }
-                return lspCompletion;
-            })
-            .concat(this.cssService.getCompletions(cleanDocument, position));
+            } else if (stComp.triggerSignature) {
+                lspCompletion.command = Command.create(
+                    'additional',
+                    'editor.action.triggerParameterHints'
+                );
+            }
+            if (!groupedCompletions.has(lspCompletion.label)) {
```

Comment:
I guess this `if` could be moved to line 301.

Reply:
moved up